crates/oxc_cli/src/main.rs

```diff
@@ -9,15 +9,42 @@ static GLOBAL: jemallocator::Jemalloc = jemallocator::Jemalloc;
 static GLOBAL: mimalloc::MiMalloc = mimalloc::MiMalloc;
 
 use std::path::PathBuf;
+use std::process::{Termination, ExitCode};
 
-use oxc_cli::{Cli, Command};
+use oxc_cli::{Cli, Command, LintResult};
 
-fn main() {
+#[derive(Debug)]
+pub enum CliRunResult {
+    None,
+    PathNotFound {
+        path: PathBuf,
+    },
+    LintResult(Vec<LintResult>),
+}
+
+impl Termination for CliRunResult {
+  fn report(self) -> ExitCode {
+      match self {
+          Self::None => ExitCode::from(0),
+          Self::PathNotFound {..} | Self::LintResult(..) => ExitCode::from(1)
+      }
+  }
+}
+
+fn main() -> CliRunResult {
     match Command::new().build().get_matches().subcommand() {
         Some(("lint", matches)) => {
             let path = matches.get_one::<PathBuf>("path").unwrap();
-            Cli::lint(path);
+            let result = Cli::lint(path);
+            let files_checked = result.as_ref().map_or(0, Vec::len);
+            println!("Checked {} files", files_checked);
+
+            if files_checked == 0 {
+              return CliRunResult::PathNotFound { path: path.clone() };
```

Comment:
I guess we can improve this a little bit more, currently if a path wasnt found, it'll just say "Checked 0 files", but it doesnt tell you that the file wasnt actually found. I can update that in this PR if you agree

Reply:
Yes we should check the path, https://doc.rust-lang.org/std/path/struct.Path.html#method.canonicalize seems like a good start.